src/views/Restoration.vue

```diff
@@ -225,10 +231,17 @@ export default class Restoration extends Vue {
       // set Applicant info in entitySnapshot
       const stateFiling = entitySnapshot.businessInfo.stateFiling
       const filing = stateFiling && await LegalServices.fetchFiling(stateFiling)
-      entitySnapshot.orgPersons = filing.restoration.parties as OrgPersonIF
-      entitySnapshot.orgPersons.forEach(o => {
-        o.officer.id = uuidv4()
-      })
+      if (!filing) {
+        throw new Error(`Invalid fetched stateFiling = ${this.getBusinessId}`)
+      } else {
+        const parties = filing.restoration.parties
+        // filter applicant from fetched parties
+        const applicant = parties.filter(
+          orgPerson => orgPerson.roles.some(role => role.roleType === RoleTypes.APPLICANT)
```

Comment:
You could also use `find` to get the first applicant in the array. It might simplify a few things for you.